### PR TITLE
csv_string_model validations

### PR DESCRIPTION
**Pull request by s12chung**
See README changes.

Most best explanation here: https://github.com/FinalCAD/csv_row_model/blob/attributes/README.md#validations

### CODE REVIEW

File: lib/csv_row_model/coercer.rb [hunk 1]
```diff
@@ -1,32 +0,0 @@
...
-      # inspired by https://github.com/MrJoy/to_bool/blob/5c9ed38e47c638725e33530ea1a8aec96281af20/lib/to_bool.rb#L23
-      Boolean => ->(s) { s =~ /^(false|f|no|n|0|)$/i ? false : true },
-      String  => ->(s) { s },
-      Integer => ->(s) { s.to_i },
-      Float   => ->(s) { s.to_f },
-      Date    => ->(s) { s.present? ? Date.parse(s) : s }
-    }
-
-  end
-end
```
Comment (s12chung) [inline lib/csv_row_model/coercer.rb:32]: @joel sorry that i removed this..... i thought about it..... the dependencies made me feel like i should change it. what i mean is... this class has 3 dependencies: `options[:type], options[:parse], caller`:
- `caller` is only required in 1 line.... so why have this here
- `options` is from `Import`... if `Import` changes the name of `:type` or `:parse`, then we have to change it here, it isn't obvious....

So... to make it more independent, i think it's better like this:

``` ruby
module CsvRowModel
  class Coercer < Struct.new(:parse, :type)
    # @return [Lambda, Proc] returns the Lambda/Proc given in the parse option or:
    # ->(original_value) { parse_proc_exists? ? parsed_value : original_value  }
    def parse_lambda
      raise ArgumentError.new("You need either :parse OR :type but not both of them") if parse && type
      parse_lambda = parse || CLASS_TO_PARSE_LAMBDA[type]
      return parse_lambda if parse_lambda
      raise ArgumentError.new("type must be #{CLASS_TO_PARSE_LAMBDA.keys.reject{|e|e.nil?}.join(', ')}")
    end

    CLASS_TO_PARSE_LAMBDA = {
      nil => ->(s) { s },
     .....
    }
  end
end
```

but... `"You need either :parse OR :type but not both of them"` implies that `:parse` and `:type` is an option too.

you kinda see what i mean...?

If it's an independent class like a validator... it makes total sense to me. i took this out to the `Attributes` module. it seemed like a good grouping.

i felt bad changing it -_-

File: lib/csv_row_model/validators/boolean_format.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class BooleanFormatValidator < ActiveModel::EachValidator # :nodoc:
```
Comment (ale7714) [inline lib/csv_row_model/validators/boolean_format.rb:1]: :+1:

**Comment by ale7714** [comment]
:white_check_mark: read

File: lib/csv_row_model/coercer.rb (hunk 1 above)
Comment (joel) [inline lib/csv_row_model/coercer.rb:32]: :smirk:

**Pull request merged**